localstack/services/kinesis/provider.py

```diff
@@ -48,19 +49,35 @@ def find_stream_for_consumer(consumer_arn):
 
 
 class KinesisProvider(KinesisApi, ServiceLifecycleHook):
+
+    server_manager: KinesisServerManager
+
+    def __init__(self):
+        self.server_manager = KinesisServerManager()
+
     def accept_state_visitor(self, visitor: StateVisitor):
         visitor.visit(kinesis_stores)
         visitor.visit(AssetDirectory(os.path.join(config.dirs.data, "kinesis")))
 
-    @staticmethod
-    def get_store(account_id: str, region_name: str) -> KinesisStore:
-        return kinesis_stores[account_id][region_name]
+    def on_before_state_load(self):
+        # no need to restart servers, since that happens lazily in `server_manager.get_server_for_account`.
+        self.server_manager.shutdown_all()
+
+    def on_before_state_reset(self):
+        self.server_manager.shutdown_all()
+
+    def on_before_stop(self):
+        self.server_manager.shutdown_all()
 
     def get_forward_url(self):
         """Return the URL of the backend Kinesis server to forward requests to"""
         account_id = get_aws_account_id()
-        starter.start_kinesis(account_id=account_id)
-        return f"http://{LOCALHOST}:{starter.get_server(account_id).port}"
+        server = self.server_manager.get_server_for_account(account_id)
+        return f"http://{LOCALHOST}:{server.port}"
+
+    @staticmethod
+    def get_store(account_id: str, region_name: str) -> KinesisStore:
```

Comment:
Had a quick look at the kinesis stores. Are we using them somewhere?

Reply:
good question, i didn't actually look into it. but may be useful to keep it